src/clib/lib/config/conf.cpp

```diff
@@ -15,53 +14,40 @@ namespace fs = std::filesystem;
 
 static auto logger = ert::get_logger("config");
 
-void conf_class_free(conf_class_type *conf_class) {
+static void conf_class_free(conf_class_type *conf_class) {
     free(conf_class->class_name);
     free(conf_class->help);
-    hash_free(conf_class->sub_classes);
-    hash_free(conf_class->item_specs);
-    vector_free(conf_class->item_mutexes);
-    free(conf_class);
+    delete conf_class;
 }
 
-void conf_class_free__(void *conf_class) {
-    conf_class_free((conf_class_type *)conf_class);
-}
-
-conf_class_type *conf_class_alloc_empty(const char *class_name,
-                                        bool require_instance, bool singleton,
-                                        const char *help) {
+std::shared_ptr<conf_class_type> make_conf_class(const char *class_name,
+                                                 bool require_instance,
+                                                 bool singleton,
+                                                 const char *help) {
     assert(class_name != NULL);
 
-    conf_class_type *conf_class =
-        (conf_class_type *)util_malloc(sizeof *conf_class);
+    auto conf_class = new conf_class_type;
 
-    conf_class->super_class = NULL;
+    conf_class->super_class = nullptr;
     conf_class->class_name = util_alloc_string_copy(class_name);
-    conf_class->help = NULL;
+    conf_class->help = nullptr;
     conf_class->require_instance = require_instance;
     conf_class->singleton = singleton;
-    conf_class->sub_classes = hash_alloc();
-    conf_class->item_specs = hash_alloc();
-    conf_class->item_mutexes = vector_alloc_new();
 
-    conf_class_set_help(conf_class, help);
-    return conf_class;
+    std::shared_ptr<conf_class_type> result(conf_class, conf_class_free);
```

Comment:
The usual way to do this is to go by using the make_unique or make_shared functions. But this is not trivial when specifying custom deleter functions.
I think how you resolve this is the way to go currently. :+1: